File: docs/sistema-de-aleatorizacion/php/sistema-de-aleatorizacion.php
<?php

class GeneradorAleatorio {
    private $abecedario = 'abcdefghijklmnopqrstuvwxyz';
    private $vocales = 'aeiou';
    private $consonantes = 'bcdfghjklmnpqrstvwxyz';

    public function numeroAleatorio($minimo, $maximo) {
        return rand($minimo, $maximo);
    }

    public function textoAleatorio($longitud) {
        $texto = '';
        for ($i = 0; $i < $longitud; $i++) {
            $indice = $this->numeroAleatorio(0, strlen($this->abecedario) - 1);
            $texto .= $this->abecedario[$indice];
        }
        return $texto;
    }

    public function palabraAleatoria($numero) {
        $palabra = '';
        for ($i = 0; $i < $numero; $i++) {
            $consonante = $this->consonantes[$this->numeroAleatorio(0, strlen($this->consonantes) - 1)];
            $vocal = $this->vocales[$this->numeroAleatorio(0, strlen($this->vocales) - 1)];
            $palabra .= $consonante . $vocal;
        }
        return $palabra;
    }

    public function literaturaAleatoria($numero) {
        $literatura = '';
        for ($i = 0; $i < $numero; $i++) {
            $literatura .= $this->palabraAleatoria($this->numeroAleatorio(1, 5)) . ' ';
        }
        return trim($literatura);
    }

    public function fechaAleatoria($anioInicio, $anioFin) {
        $año = $this->numeroAleatorio($anioInicio, $anioFin);
        $mes = $this->numeroAleatorio(1, 12);
        $dia = $this->numeroAleatorio(1, cal_days_in_month(CAL_GREGORIAN, $mes, $año));

        $horas = $this->numeroAleatorio(0, 23);
        $minutos = $this->numeroAleatorio(0, 59);
        $segundos = $this->numeroAleatorio(0, 59);
        $milisegundos = $this->numeroAleatorio(0, 999);

        return new DateTime("$año-$mes-$dia $horas:$minutos:$segundos.$milisegundos");
    }
}

// Uso
$generador = new GeneradorAleatorio();

echo 'Número Aleatorio: ' . $generador->numeroAleatorio(1, 10) . PHP_EOL;
echo 'Texto Aleatorio: ' . $generador->textoAleatorio(8) . PHP_EOL;
echo 'Palabra Aleatoria: ' . $generador->palabraAleatoria(3) . PHP_EOL;
echo 'Literatura Aleatoria: ' . $generador->literaturaAleatoria(5) . PHP_EOL;
echo 'Fecha Aleatoria: ' . $generador->fechaAleatoria(2000, 2022)->format('Y-m-d H:i:s.u') . PHP_EOL;
?>

File: sistema-de-aleatorizacion/ruby/sistema-de-aleatorizacion.rb
class GeneradorAleatorio
    def initialize
      @abecedario = 'abcdefghijklmnopqrstuvwxyz'
      @vocales = 'aeiou'
      @consonantes = 'bcdfghjklmnpqrstvwxyz'
    end
  
    def numero_aleatorio(minimo, maximo)
      rand(minimo..maximo)
    end
  
    def texto_aleatorio(longitud)
      texto = ''
      longitud.times { texto += @abecedario[rand(@abecedario.length)] }
      texto
    end
  
    def palabra_aleatoria(numero)
      palabra = ''
      numero.times do
        consonante = @consonantes[rand(@consonantes.length)]
        vocal = @vocales[rand(@vocales.length)]
        palabra += "#{consonante}#{vocal}"
      end
      palabra
    end
  
    def literatura_aleatoria(numero)
      literatura = ''
      numero.times { literatura += "#{palabra_aleatoria(numero_aleatorio(1, 5))} " }
      literatura.strip
    end
  
    def fecha_aleatoria(anio_inicio, anio_fin)
      año = numero_aleatorio(anio_inicio, anio_fin)
      mes = numero_aleatorio(1, 12)
      dia = numero_aleatorio(1, Date.new(año, mes, -1).day)
  
      horas = numero_aleatorio(0, 23)
      minutos = numero_aleatorio(0, 59)
      segundos = numero_aleatorio(0, 59)
      milisegundos = numero_aleatorio(0, 999)
  
      Time.new(año, mes, dia, horas, minutos, segundos, milisegundos * 1000)
    end
  end
  
  # Uso
  generador = GeneradorAleatorio.new
  
  puts 'Número Aleatorio:', generador.numero_aleatorio(1, 10)
  puts 'Texto Aleatorio:', generador.texto_aleatorio(8)
  puts 'Palabra Aleatoria:', generador.palabra_aleatoria(3)
  puts 'Literatura Aleatoria:', generador.literatura_aleatoria(5)
  puts 'Fecha Aleatoria:', generador.fecha_aleatoria(2000, 2022)
  